src/csvcubed/writers/qbwriter.py

```diff
@@ -373,8 +495,33 @@ def _generate_qb_dataset_dsd_definitions(self) -> QbDataSetInCatalog:
 
                 dataset.structure.components |= set(component_specs_for_col)
 
+        if self.is_cube_in_pivoted_shape:
+            dataset.structure.sliceKey.add(self._get_cross_measures_slice_key())
+
         return dataset
 
+    def _get_cross_measures_slice_key(self) -> rdf.qb.SliceKey:
+        # Setting up Slice Key for slices which range over measures.
+        slice_key = rdf.qb.SliceKey(
+            self._new_uri_helper.get_slice_key_across_measures_uri()
+        )
+
+        for dimension_column in get_columns_of_dsd_type(self.cube, QbDimension):
+            dimension = dimension_column.structural_definition
+            dimension_uri: str
+            if isinstance(dimension, NewQbDimension):
+                dimension_uri = self._new_uri_helper.get_dimension_uri(
+                    dimension.uri_safe_identifier
+                )
+            elif isinstance(dimension, ExistingQbDimension):
+                dimension_uri = dimension.dimension_uri
+            else:
+                raise ValueError(f"Unmatched QbDimension type {type(dimension)}")
```

Comment:
Again, can we use `TypeError` here? It's a more accurate exception to raise than the one I stuck in here.